docs/EdgeApps.md

```diff
@@ -416,6 +416,140 @@ settings:
+
+- **Schema**: The JSON must include `schema_version` and a `properties` object.
+- **Common keys**:
+  - `properties.type`: One of `datetime`, `number`, `select`, `boolean`, `textarea`, `url`.
+  - `properties.help_text`: Human-friendly description shown in the UI.
+  - `properties.options` (only for `select`): Array of `{ label, value }` options.
+- **Storage**: Use `type: string` for all non-secret fields; use `type: secret` for password-like fields. The UI will coerce values appropriately (e.g., booleans) but values are stored as strings unless `type: secret`.
+- **Defaults**: Provide `default_value` at the setting level. For booleans, use `'true'` or `'false'` as strings.
+
+Examples:
+
+**Datetime picker**
+
+```yaml
+settings:
+  date_time_field:
+    type: string
+    title: Start Date Time
+    optional: false
+    help_text: |
+      {
+        "schema_version": 1,
+        "properties": {
+          "help_text": "The start date and time of the event",
+          "type": "datetime"
+        }
+      }
+```
+
+**Number input**
+
+```yaml
+settings:
+  number_field:
+    type: string
+    title: Attendee Count
+    optional: false
+    help_text: |
+      {
+        "schema_version": 1,
+        "properties": {
+          "help_text": "The expected count of attendees",
+          "type": "number"
+        }
+      }
+```
+
+**Select (dropdown) input**
+
+```yaml
+settings:
+  select_field:
+    type: string
+    title: Select Role
+    default_value: editor
+    optional: false
+    help_text: |
+      {
+        "schema_version": 1,
+        "properties": {
+          "type": "select",
+          "help_text": "The role of the user",
+          "options": [
+            { "label": "Admin",  "value": "admin" },
+            { "label": "Editor", "value": "editor" },
+            { "label": "Viewer", "value": "viewer" }
+          ]
+        }
+      }
+```
+
+**Boolean (switch) input**
+
+```yaml
+settings:
+  switch_field:
+    type: string
+    title: Subscribe
+    default_value: "true" # or 'false'
+    optional: false
+    help_text: |
+      {
+        "schema_version": 1,
+        "properties": {
+          "help_text": "Subscribe to updates",
+          "type": "boolean"
+        }
+      }
+```
+
+**Text area input**
+
+```yaml
+settings:
+  text_area_field:
+    type: string
+    title: Description
+    default_value: Field description
+    optional: false
+    help_text: |
+      {
+        "schema_version": 1,
+        "properties": {
+          "help_text": "The description of the event",
+          "type": "textarea"
+        }
+      }
+```
+
+**URL input**
+
+```yaml
+settings:
+  url_field:
+    type: string
+    title: Website URL
+    optional: false
+    help_text: |
+      {
+        "schema_version": 1,
+        "properties": {
+          "help_text": "The URL of the website",
+          "type": "url"
+        }
+      }
```

Comment:
I find it pretty inconvenient. 
Could we deal with the type of the `help_text` internally, and in allow the data to be a dict in yaml? 
```suggestion
    help_text: 
      schema_version: 1,
      properties: 
        help_text": "The URL of the website",
        type: "url"
```

Reply:
@renatgalimov this is indeed more native. I don't live the idea of mixing YAML and JSON either. Agree that this is a better approach.

Reply:
@iamplugged let's land this PR and create a ticket to change help_text schema from JSON to YAML

Reply:
https://phorge.wireload.net/T9633